packages/core/core/src/services/document-service/utils/populate.ts

```diff
@@ -0,0 +1,51 @@
+import { Common } from '@strapi/types';
+import { contentTypes } from '@strapi/utils';
+
+// We want to build a populate object based on the schema
+export const getDeepPopulate = (uid: Common.UID.Schema) => {
+  const model = strapi.getModel(uid);
+  const attributes = Object.entries(model.attributes);
+
+  return attributes.reduce((acc: any, [attributeName, attribute]) => {
+    switch (attribute.type) {
+      case 'relation': {
+        // TODO: Should this just be a plain list?
+        // Ignore createdBy, updatedBy, ...
+        const isVisible = contentTypes.isVisibleAttribute(model, attributeName);
+        if (isVisible) {
+          acc[attributeName] = { fields: ['document_id', 'locale'] };
+        }
+        break;
+      }
+
+      case 'media': {
+        acc[attributeName] = { fields: ['id'] };
```

Comment:
Do we need to worry about if a media attribute is private?

Reply:
in this case I believe it's not important, as none of this data is returned to the user, rather this populate is used to gather all data and copy it over (e.g. publishing)